pkg/router/controller/route_secret_manager_test.go

```diff
@@ -0,0 +1,1148 @@
+		{
+			name: "route updated: old route without externalCertificate, new route without externalCertificate",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{},
+			},
+			secretManager: fake.SecretManager{
+				IsRegistered: false,
+			},
+			eventType: watch.Modified,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{},
+			},
+			expectedEventType: watch.Modified,
+		},
+
+		// scenarios when route is deleted
+		{
+			name: "route deleted without externalCertificate registered",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{},
+			},
+			secretManager: fake.SecretManager{IsRegistered: false},
+			eventType:     watch.Deleted,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{},
+			},
+			expectedEventType: watch.Deleted,
+		},
+		{
+			name: "route deleted with externalCertificate registered",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{IsRegistered: true},
+			eventType:     watch.Deleted,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			expectedEventType: watch.Deleted,
+		},
+		{
+			name: "route deleted with externalCertificate registered, but got error from secretManager",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{IsRegistered: true, Err: fmt.Errorf("something")},
+			eventType:     watch.Deleted,
+			expectedError: true,
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			p := &fakePlugin{}
+			recorder := routeStatusRecorder{rejections: make(map[string]string)}
+
+			// assign default value to expectedRejections
+			if s.expectedRejections == nil {
+				s.expectedRejections = map[string]string{}
+			}
```

Comment:
If you keep `rejections` and `expectedRejections` unset `reflect.DeepEqual()` should give `true`:
```suggestion
			recorder := routeStatusRecorder{}
```

Reply:
If we keep `rejections` unset, it will cause panic 
```
panic: assignment to entry in nil map
```
So, need to create the map.